### PR TITLE
Issue #17 Slot updates update old table slots rather than marking them as removed

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

**Comment by terracotta-org** [comment]
Can one of the admins verify this patch?

**Comment by terracotta-org** [comment]
Can one of the admins verify this patch?

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
test this please

File: src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java [hunk 1]
```diff
@@ -1410,8 +1410,9 @@ private void cleanPendingTables(int hash, long encoding) {
   
           if (isTerminating(entry)) {
             break;
-          } else if (isPresent(entry) && (hash == entry.get(KEY_HASHCODE)) && (encoding == readLong(entry, ENCODING))) {
-            entry.put(STATUS, STATUS_REMOVED);
+          } else if (isPresent(entry) && (hash == entry.get(KEY_HASHCODE)) && (oldEncoding == readLong(entry, ENCODING))) {
+            entry.put(newEntry);
+            newEntry.flip();
```
Comment (cljohnso) [inline src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java]: Is this `flip()` necessary or proper?  The calls to `updatePendingTables` from `slotUpdated` all have "flipped" buffers.

**Pull request merged**